qiskit/algorithms/minimum_eigensolvers/adapt_vqe.py

```diff
@@ -85,38 +87,79 @@ class AdaptVQE(VariationalAlgorithm, MinimumEigensolver):
+            the algorithm has converged and terminates.
+        eigenvalue_threshold: once the eigenvalue has changed by less than this threshold from one
+            iteration to the next, the algorithm has converged and terminates.
         max_iterations: the maximum number of iterations for the adaptive loop. If ``None``, the
             algorithm is not bound in its number of iterations.
     """
 
+    @deprecate_arg(
+        "threshold",
+        since="0.24.0",
+        pending=True,
+        new_alias="gradient_threshold",
+    )
     def __init__(
         self,
         solver: VQE,
         *,
-        threshold: float = 1e-5,
+        gradient_threshold: float = 1e-5,
+        eigenvalue_threshold: float = 1e-5,
         max_iterations: int | None = None,
+        threshold: float | None = None,  # pylint: disable=unused-argument
     ) -> None:
         """
         Args:
             solver: a :class:`~.VQE` instance used internally to compute the minimum eigenvalues.
                 It is a requirement that the :attr:`~.VQE.ansatz` of this solver is of type
                 :class:`qiskit.circuit.library.EvolvedOperatorAnsatz`.
-            threshold: the convergence threshold for the algorithm. Once all gradients have an
-                absolute value smaller than this threshold, the algorithm terminates.
+            gradient_threshold: once all gradients have an absolute value smaller than this
+                threshold, the algorithm has converged and terminates.
+            eigenvalue_threshold: once the eigenvalue has changed by less than this threshold from
+                one iteration to the next, the algorithm has converged and terminates.
             max_iterations: the maximum number of iterations for the adaptive loop. If ``None``, the
                 algorithm is not bound in its number of iterations.
+            threshold: once all gradients have an absolute value smaller than this threshold, the
+                algorithm has converged and terminates. Defaults to ``1e-5``.
         """
-        validate_min("threshold", threshold, 1e-15)
+        validate_min("gradient_threshold", gradient_threshold, 1e-15)
+        validate_min("eigenvalue_threshold", eigenvalue_threshold, 1e-15)
 
         self.solver = solver
-        self.threshold = threshold
+        self.gradient_threshold = gradient_threshold
+        self.eigenvalue_threshold = eigenvalue_threshold
         self.max_iterations = max_iterations
         self._tmp_ansatz: EvolvedOperatorAnsatz | None = None
         self._excitation_pool: list[OperatorBase] = []
         self._excitation_list: list[OperatorBase] = []
 
+    @property
+    def threshold(self) -> float:
+        """The threshold for the gradients.
+
+        Once all gradients have an absolute value smaller than this threshold, the algorithm has
+        converged and terminates.
+        """
+        msg = (
```

Comment:
~The thing is though, that the `threshold` used to be a public attribute which was documented in the class docstring. There I simply renamed it now. But in order to make things work the same way I added this new `threshold` property getter and setter which internally just redirects to the new `gradient_threshold` attribute in order to be consistent. I do not think that this can be handled by the decorator injecting into the docs, can it?~

EDIT: ah I suppose you mean I could simply use the `deprecate_func` decorator on top of the new property, that makes sense :+1: